### PR TITLE
Add CRIU version check for checkpoint and restore

### PR DESCRIPTION
**Pull request by adrianreber**
As the checkpoint/restore implementation in Podman requires at least CRIU 3.11, this vendors in the Go bindings of CRIU and uses these bindings to check the version of the installed CRIU binary.

### CODE REVIEW

File: libpod/container_internal_linux.go [hunk 1]
```diff
@@ -355,6 +360,10 @@ func (c *Container) checkpoint(ctx context.Context, keep bool) (err error) {
 
 func (c *Container) restore(ctx context.Context, keep bool) (err error) {
 
+	if !criu.CheckForCriu() {
+		return errors.Errorf("checkpointing a container requires at least CRIU 3.11")
```
Comment (mheon) [inline libpod/container_internal_linux.go]: Change to "restoring a container" instead of "checkpointing a container"

File: pkg/criu/criu.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import (
+	"github.com/checkpoint-restore/go-criu"
+)
+
+// CheckForCriu uses CRIU's go bindings to check if the CRIU
+// binary exists and if it at least the version Podman needs.
+func CheckForCriu() bool {
+	c := criu.MakeCriu()
+	// For Podman at least CRIU 3.11 is required
+	result, err := c.IsCriuAtLeast(31100)
```
Comment (mheon) [inline pkg/criu/criu.go]: Nit - version number would probably be better as a package constant

File: libpod/container_internal_linux.go (hunk 1 above)
Comment (adrianreber) [reply libpod/container_internal_linux.go]: Ah, copy/paste. Will fix.

**Comment by mheon** [comment]
@adrianreber We just merged something that should fix CI, mind rebasing?

**Comment by adrianreber** [comment]
@mheon thanks for the review. I changed the things you pointed out and force pushed.

I now get the following if CRIU is too old:

```
#  podman container checkpoint --keep podman-criu-test 
failed to checkpoint container 91b1f32f7: checkpointing a container requires at least CRIU 31100
```

**Comment by baude** [comment]
bot, retest this please

**Comment by adrianreber** [comment]
Yes, the vendoring is not correct, need to fix that.

**Comment by adrianreber** [comment]
I just saw that CRIU is no longer installed in the new test environment. All checkpoint/restore test cases are now skipped.

**Comment by rhatdan** [comment]
lgtm

**Comment by mheon** [comment]
Hm. I'll talk to Chris and Brent and see about getting CRIU on our test VMs
in Cirrus

**Comment by adrianreber** [comment]
> Hm. I'll talk to Chris and Brent and see about getting CRIU on our test VMs in Cirrus

The necessary steps are still in the Dockerfile.

**Comment by mheon** [comment]
A lot of our tests are now on bare VMs, so those won't run the Dockerfile -
we'll need it added there. Shouldn't be that bad.

**Comment by baude** [comment]
For F28 in PAPR, if the packaged version is sufficient, then an edit here would do it: -> https://github.com/containers/libpod/blob/master/.papr.yml#L112

@cevich can you say where we can add additional packages for cirrus?

**Comment by cevich** [comment]
@baude currently all package installing happens at image-build time in ``contrib/cirrus/packer/*.sh`` for the various distros.  However, if this is something that you expect to update frequently (more than once a week or so), it may make more sense to install it at runtime via the distro-specific ``case`` blocks in ``contrib/cirrus/setup_environment.sh``.  If neither makes sense, let me know.

**Comment by adrianreber** [comment]
CRIU would not need to be updated often. Only if a new feature is required, which Podman needs.

Currently Podman needs CRIU from git (that's what I added to the docker file when travis testing was still done). Once CRIU 3.11 is released (probably in the next few weeks), we can just install the Fedora package (for the Fedora tests), once I have updated the package there.

I would like to see checkpoint/restore testing enabled again in Podman before CRIU 3.11 is released to make sure that checkpoint/restore does not brake, that would mean installation from git.

But if you say that this is too much effort we can also wait until CRIU 3.11 is released and is part of Fedora. Whatever works best. Let me know if there is anything I can do to help here.

**Comment by mheon** [comment]
I'm fine with building from Git - we have a few other dependencies we grab this way (runc, for example, we're building a specified commit there).

This will need changes to the Cirrus and Papr scripts and the Dockerfile to install from source, but I don't think they should be that bad

**Comment by cevich** [comment]
> (that's what I added to the docker file when travis testing was still done)

Oh, that makes it even easier.  I can just copy-paste what you did for the Dockerfile, [something like this for cirrus.](https://paste.fedoraproject.org/paste/X7aZmU6zAuM8I1fovotVRQ) super easy.

**Comment by cevich** [comment]
Okay done: #1660

I added it for all distros, but should it be just for Ubuntu?

**Comment by adrianreber** [comment]
@cevich thanks, looks perfect. CRIU and podman checkpoint/restore should work on all distros.

**Comment by cevich** [comment]
@adrianreber excellent, okay I'l carry on with that then, thanks.

**Comment by adrianreber** [comment]
All automated tests are now green (except cirrus ci which still is Pending)

**Pull request merged**